### ISSUE TITLE
BHTikTok Plus region setting doesn't work

### ISSUE THREAD

**Issue by Anime-Forevere** [swaggyP36000/TrollStore-IPAs#141]
Hello.
I noticed an issue that I have from a long time ago. I think since base version >32.2.0 (32.2.0 works fine) of BhTikTok Plus region setting doesn't get applied. I'm using signulous to sign the app, old version is working fine.
iPhone 11, iOS 17.2.1

**Comment by Anime-Forevere** [comment]
I installed v32.9.1 (released yesterday) and made a screen recording, maybe it will help you see something? App was fresh installed, there are no other TikTok apps. On the video, there are only Russian videos (as it is restricted to only old Russian videos where I live) and changing region doesn't change anything (if it worked fine, I would get modern videos from other countries). 
[Here's the screen recording](https://veed.io/view/b1b53a69-ecb6-4bd1-b5a8-f0ab2422e011)

**Comment by ArianaMinajj** [comment]
Thank you for providing the screen recording. I've encountered the same problem and have dedicated considerable time to testing various versions of BHTiktok, ranging from 32.9.1 to 32.2.0, based on your indication that it functioned correctly in the latter. 

However, it seems to not be working *(for me)* even on the 32.2.0 version and below and I can't pinpoint what I'm doing incorrectly..

*(I've also attempted to use that feature on TiktokLRD and TikTokDark without any success..)*

Unfortunately, I'm at a loss for determining the root cause of the problem, so I apologize for not being able to resolve your issue!

**Comment by Anime-Forevere** [comment]
I found a solution to this bug:
1. Install TikTok and DON'T login into account
2. Change the region and apply settings (also make sure it worked for you by checking recommendations)
3. Login into your account (either normally or through "forgot password")

**Comment by ArianaMinajj** [comment]
Thank you but I'm still encountering the same issue while trying that with TrollStore following the provided instructions.

It could still be valuable for individuals who are working within the same signing environment as you!

**Comment by Anime-Forevere** [comment]
Found more info about this. 
TLDR: TikTok is changing to new SDK from a deprecated one (which causes region setting to break)

As I was told, TikTok is changing to new Apple SDK, probably to 16.4 because it made [CTCarrier](https://developer.apple.com/documentation/coretelephony/ctcarrier) class deprecated. Tweaks probably used that to trick TikTok into thinking that you are from other country, but now the system changed and it uses some other technology. I guess it is a matter of time for someone to find new ability to change region.

**Comment by ArianaMinajj** [comment]
I believe it would be beneficial to share that **[here](https://x.com/dayanch96/status/1748053528961073183?)** since dayanch96 is the maintainer of BHTiktokPlus!

**Comment by Pato2222** [comment]
> I believe it would be beneficial to share that **[here](https://x.com/dayanch96/status/1748053528961073183?)** since dayanch96 is the maintainer of BHTiktokPlus!

@Anime-Forevere you should!
